### PR TITLE
docs: add tutorials for running macaron

### LINKED ISSUE #198: Add tutorials for running Macaron
We need to add tutorials based on the Docker run script for supported inputs and scenarios:
- [x]  Using remote GitHub URL path, branch, and commit as input
- [x] Using SBOM as input
- [x] Using locally cloned repo.

### PR DESCRIPTION
**Pull request by tromai**
Close #198 .

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -129,7 +129,6 @@ repos:
   # - id: python-check-blanket-type-ignore
   - id: python-check-mock-methods
   - id: python-use-type-annotations
-  - id: rst-backticks
```
Comment (tromai) [inline .pre-commit-config.yaml:132]: This pre-commit hook mistakes a link in Sphinx
```rst
`Link title <link>`_ 
```
To be a "code" block 
```rst
``inlint_code``
```
And reports a false negative.

Comment (behnazh-w) [reply .pre-commit-config.yaml:132]: I cannot reproduce this locally if I add `rst-backticks` back. Can you post the error?

File: docs/source/pages/installation.rst [hunk 2]
```diff
@@ -0,0 +1,65 @@
+.. Copyright (c) 2023 - 2023, Oracle and/or its affiliates. All rights reserved.
+.. Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+.. _installation-guide:
+
+============
+Installation
```
Comment (behnazh-w) [inline docs/source/pages/installation.rst:7]:
```suggestion
Installation Guide
```

File: docs/source/pages/installation.rst [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+============
+
+-------------
+Prerequisites
+-------------
+- Installations of ``wget`` or ``curl`` and ``bash`` must be available and on the path.
+- Docker (or docker equivalent for your host OS) must be installed, with a docker command line equivalent to Docker 17.06 (Oracle Container Runtime 19.03) and the user that is executing a member of the operating system group ``docker`` (to run Docker in `rootless mode <https://docs.docker.com/engine/security/rootless/>`_).
```
Comment (behnazh-w) [inline docs/source/pages/installation.rst:14]:
```suggestion
- Docker (or docker equivalent for your host OS) must be installed, with a docker command line equivalent to Docker 17.06 (Oracle Container Runtime 19.03) and the user should be a member of the operating system group ``docker`` (to run Docker in `rootless mode <https://docs.docker.com/engine/security/rootless/>`_).
```

File: .pre-commit-config.yaml (hunk 1 above)
Comment (tromai) [reply .pre-commit-config.yaml:132]: Hmmm you are right. I think I made some mistakes. I try using this hook in the latest commit and it works fine. I will revert the changes.

File: docs/source/pages/installation.rst (hunk 3 above)
Comment (tromai) [reply docs/source/pages/installation.rst]: Fixed in https://github.com/oracle-samples/macaron/pull/232/commits/11976b13ad452d9a8c94e09cecb9aeca5c34e004

File: docs/source/pages/installation.rst (hunk 2 above)
Comment (tromai) [reply docs/source/pages/installation.rst]: Fixed in https://github.com/oracle-samples/macaron/pull/232/commits/11976b13ad452d9a8c94e09cecb9aeca5c34e004

File: .pre-commit-config.yaml (hunk 1 above)
Comment (tromai) [reply .pre-commit-config.yaml]: Reverted in https://github.com/oracle-samples/macaron/pull/232/commits/11976b13ad452d9a8c94e09cecb9aeca5c34e004

**Comment by behnazh-w** [comment]
I wonder if we should use expected reports that we use for integration tests as examples in docs to ensure they are always up-to-date :thinking:

**Comment by tromai** [comment]
I think it's best to keep the example reports up to date automatically. However, I am thinking how could we tell the Sphinx engine to copy the expected reports to the built directory for the final documentation artifact or whether it would complicate our setup :thinking:?

**Comment by behnazh-w** [comment]
> I think it's best to keep the example reports up to date automatically. However, I am thinking how could we tell the Sphinx engine to copy the expected reports to the built directory for the final documentation artifact or whether it would complicate our setup thinking?

We could add relevant tests in integration tests where expected output is read from these example reports to make sure they are always up-to-date. But we should be careful to not add too many tests as the integration tests are already long. We could also modify some of the existing tests to read the expected output from these sample reports.

I'm happy to merge it as it is and track this in a separate issue and PR.

**Comment by behnazh-w** [comment]
> I'm happy to merge it as it is and track this in a separate issue and PR.

Can you please add an issue to keep track of it?

**Review by behnazh-w — APPROVED** [review]

**Pull request merged**